homeassistant/components/rflink.py

```diff
@@ -166,15 +167,17 @@ def event_callback(event):
         if entities:
             # Propagate event to every entity matching the device id
             for entity in entities:
-                _LOGGER.debug('passing event to %s', entities)
-                entity.handle_event(event)
+                _LOGGER.debug('passing event to %s', entity)
+                async_dispatcher_send(hass, SIGNAL_HANDLE_EVENT, entity, event)
         else:
-            _LOGGER.debug('device_id not known, adding new device')
-
             # If device is not yet known, register with platform (if loaded)
             if event_type in hass.data[DATA_DEVICE_REGISTER]:
+                _LOGGER.debug('device_id not known, adding new device')
+                # TODO: Is there a race condition here if we get another event before async_run_job is done?
                 hass.async_run_job(
```

Comment:
Why do we use run job? Are we calling coroutines sometimes and callbacks sometimes?

Reply:
What do you suggest to use otherwise?

Reply:
It depends on what the function is that we want to call. If it's always a coroutine we should schedule it with `hass.async_create_task`. If it's always a callback, we can call it directly. If it's a mix, we can use `hass.async_run_job`.